web/docs/language/features.md

```diff
@@ -602,6 +602,193 @@ try {
+const updateAction = Tasks.update.useAction()
+const deleteAction = Tasks.delete.useAction()
+
+// The CRUD operations are using the existing actions and queries
+// under the hood, so all the options are available as before.
+```
+
+We'll show usage in the example below.
+
+#### Full example
+
+Let's say we have an entity called `Task` in our Wasp file:
+```wasp title="main.wasp"
+// ...
+
+entity Task {=psl
+  id Int @id @default(autoincrement())
+  description String
+  isDone Boolean
+psl=}
+```
+
+We would then add the following to our Wasp file to enable automatic CRUD for `Task`:
+
+```wasp title="main.wasp"
+// ...
+
+crud Tasks {
+  entity: Task,
+  operations: {
+    getAll: {
+      isPublic: true,
+    },
+    get: {},
+    create: {
+      overrideFn: import { createTask } from "@server/tasks.js",
+    },
+    update: {},
+  },
+}
+```
+
+And let's say we have the following implementation in `@server/tasks.js`:
+
+```ts title="src/server/tasks.js"
+import type { CreateAction } from '@wasp/crud/Tasks'
+import type { Task } from '@wasp/entities'
+
+export const createTask: CreateAction<{
+  description: string;
+  isDone: boolean;
+}, Task> = async (args, context) => {
+  // Let's say we have this requirement that only user with id 1 can create tasks.
+  if (!context.user || context.user.id !== 1) {
+    throw new Error('Only user with id 1 can create tasks.')
+  }
+
+  const { description, isDone } = args.data
+  const { Task } = context.entities
+
+  return await Task.create({
+    data: {
+      description,
+      isDone,
+    },
+  })
+}
+```
+
+- We defined a new CRUD called `Tasks` which uses the `Task` entity to perform all the operations.
+- We omitted the `delete` operation, so the `delete` operation will not be generated. We also specified that the `getAll` operation is public, meaning that it can be called without authentication.
+- We also specified an override function for the `create` operation. This means that the `create` operation will not be generated, but instead, the `createTask` function from `@server/tasks.js` will be used.
+
+And let's use the generated operations in our client code:
+
+```jsx title="pages/TasksList.jsx"
+import { Tasks } from '@wasp/crud/Tasks'
+
+const TasksList = () => {
+  const { data: tasks, isLoading, error } = Tasks.getAll.useQuery()
+  // const { data: task } = Tasks.get.useQuery({ id: 1 })
+
+  const createAction = Tasks.create.useAction()
+  // const updateAction = Tasks.update.useAction()
+  // const deleteAction = Tasks.delete.useAction()
+
+  function handleCreateTask() {
+    createAction({ description: 'New task', isDone: false })
+  }
+
+  if (isLoading) return <div>Loading...</div>
+  if (error) return <div>Error: {error.message}</div>
+  return (
+    <div>
+      <button onClick={handleCreateTask}>Create task</button>
+      <ul>
+        {tasks.map(task => (
+          <li key={task.id}>{task.description}</li>
+        ))}
+      </ul>
+    </div>
+  )
+}
+```
+
```

Comment:
Some stuff is missing:
1. Explanation on how automatically generated code works + warning that right now it accepts all of the fields of the entity when doing create / update, which can be troublesome, we need to warn about that. If that is an issue, they should override.
2. Future: validation (which will solve the issue above), auth.
3. Mention this is experimental feature (mainly due to (1), and our plans explained in (2)).